src/classes/COFF_Affiliation_TDTM.cls

```diff
@@ -357,21 +360,25 @@ public class COFF_Affiliation_TDTM extends TDTM_Runnable {
         Id facultyRecordTypeId = UTIL_Describe.getFacultyConnectionRecType();
         Map<Id, Map<Id, Course_Enrollment__c>> offeringFacultyConnIds = getOfferingIdFacultyIdConnection(offsWithChangedFaculty, facultyRecordTypeId);
         String queryString = 'SELECT Id, Affiliation__c, Account__c, Course_Offering__c, Primary__c FROM Course_Enrollment__c';
-        queryString += ' WHERE Contact__c in :outdatedFacultyIDs';
-        queryString += ' and Course_Offering__c in :offsWithChangedFaculty';
+        queryString += ' WHERE Contact__c IN :outdatedFacultyIDs';
+        queryString += ' AND Course_Offering__c IN :offsWithChangedFaculty';
         //Record type of course enrollment not in the package
         if (UTIL_Describe.isValidField(UTIL_Namespace.StrTokenNSPrefix('Course_Enrollment__c'),'RecordTypeId') && facultyRecordTypeId != null) {
-            queryString += ' and RecordTypeId = :facultyRecordTypeId';
+            queryString += ' AND RecordTypeId = :facultyRecordTypeId';
         }
         List<Course_Enrollment__c> oldConns = Database.query(queryString);
-        for(Integer i = 0; i < oldConns.size(); i++) {
+
+        for (Integer i = 0; i < oldConns.size(); i++) {
             if (offeringFacultyConnIds.containsKey(oldConns[i].Course_Offering__c)
                     && offeringFacultyConnIds.get(oldConns[i].Course_Offering__c) != null
                     && offeringFacultyConnIds.get(oldConns[i].Course_Offering__c).containsKey(updatedFacultyIDs[i])) {
+                
                 Course_Enrollment__c existingConn = offeringFacultyConnIds.get(oldConns[i].Course_Offering__c).get(updatedFacultyIDs[i]);
-                existingConn.Primary__c = true;
-                // Current Course Connection exists on Course Offering with Contact, mark as Primary
-                objectsToUpdate.add((SObject)existingConn);
+                if (!existingConn.Primary__c) {
```

Comment:
I've made some efficiency improvements here and further down in other methods. When testing the reentrancy prevention, I realized that this class was updating course connections even when the Primary field was already set to the correct value.